common/src/main/java/com/datastrato/gravitino/dto/rel/indexes/IndexDTO.java

```diff
@@ -0,0 +1,73 @@
+/*
+ * Copyright 2024 Datastrato Pvt Ltd.
+ * This software is licensed under the Apache License version 2.
+ */
+package com.datastrato.gravitino.dto.rel.indexes;
+
+import com.datastrato.gravitino.rel.expressions.Indexes.Index;
+import com.google.common.base.Preconditions;
+
+public class IndexDTO implements Index {
```

Comment:
IndexDTO is only used for transimission. We should have a IndexImpl like what @mchades did in API module.